### ISSUE TITLE
Incorrect handling of UsersJoined

### ISSUE THREAD

**Issue by dexter3k** [tucnak/telebot#365]
This piece of code
https://github.com/tucnak/telebot/blob/7a9c471cf32c81870f1bad0fdfc8fa6a646abb89/bot.go#L263
Incorrectly handles UserJoined logic.

When multiple users are added to the chat, UsersJoined will contain list of users joined, but also UserJoined will also contain one of the users from the list. Currently because
```golang
if m.UserJoined != nil {
	b.handle(OnUserJoined, m)
	return
}

if m.UsersJoined != nil {
	for _, user := range m.UsersJoined {
		m.UserJoined = &user
		b.handle(OnUserJoined, m)
	}
	return
}
```
is there, only the first user will be handled.
One should probably use something like this:
```golang
if m.UsersJoined != nil {
	for _, user := range m.UsersJoined {
		m.UserJoined = &user
		b.handle(OnUserJoined, m)
	}
	return
} else if m.UserJoined != nil {
	b.handle(OnUserJoined, m)
	return
}
```

**Comment by dexter3k** [comment]
Now, there seems to be another issue closely related to this, not sure if this better be another issue..

There is a data race in your code here
```golang
for _, user := range m.UsersJoined {
  m.UserJoined = &user
  b.handle(OnUserJoined, m)
}
```

Because b.handle runs handlers asynchronously in some cases, the data at &user will get updated to the last user in the list. Thus all, or some calls to handlers will get same user.

You can try live example of this problem simplified here: https://play.golang.org/p/DbOZgUOzyV1

**Comment by tucnak** [comment]
Fixed in #366
